### PR TITLE
[Merged by Bors] - chore: remove the `Smooth` alias for infinitely differentiable functions between manifolds

### PR DESCRIPTION
**Pull request by sgouezel**
The abbreviation `Smooth` for `ContMDiff` with infinite smoothness had been introduced in mathlib in ancient times. This has lead to a big API duplication, for no visible gain. We deprecate this abbreviation (with its friends `SmoothAt` and so on) and all the lemmas involving it.

In the longer run, we might consider reintroducing it as a mere notation (without any specific API), but this is not part of this PR, devoted purely to the deprecation. No mathematical content added.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
I've reviewed the changes to the `MFDeriv`, `Sheaf` and `VectorBundle` directories: only very minor questions so far. Two larger points worth flagging:
- would you like to remove the now-used `abbrev`s `Smooth` etc?
- the deprecation dates in your lemmas are slightly inconsistent: not a big deal, in any case

File: Mathlib/Geometry/Manifold/VectorBundle/Basic.lean [hunk 1]
```diff
@@ -284,27 +281,23 @@ section SmoothCoordChange
 variable {F E}
 variable (e e' : Trivialization F (π F E)) [MemTrivializationAtlas e] [MemTrivializationAtlas e']
 
-theorem smoothOn_coordChangeL :
-    SmoothOn IB 𝓘(𝕜, F →L[𝕜] F) (fun b : B => (e.coordChangeL 𝕜 e' b : F →L[𝕜] F))
+theorem contMDiffOn_coordChangeL :
+    ContMDiffOn IB 𝓘(𝕜, F →L[𝕜] F) n (fun b : B => (e.coordChangeL 𝕜 e' b : F →L[𝕜] F))
```
[the remark this answers is not in the document]
Comment (sgouezel) [reply Mathlib/Geometry/Manifold/VectorBundle/Basic.lean:285]: The version with a general `n` is stronger than the version with `top` (well, they are equivalent, but often the version with a general implicit `n` is more convenient, for instance when chaining).

File: Mathlib/Geometry/Manifold/MFDeriv/Basic.lean [hunk 2]
```diff
@@ -477,36 +477,39 @@ theorem ContMDiff.mdifferentiableAt (hf : ContMDiff I I' n f) (hn : 1 ≤ n) :
     MDifferentiableAt I I' f x :=
   hf.contMDiffAt.mdifferentiableAt hn
 
+theorem ContMDiff.mdifferentiableWithinAt (hf : ContMDiff I I' n f) (hn : 1 ≤ n) :
```
[the remark this answers is not in the document]
Comment (sgouezel) [reply Mathlib/Geometry/Manifold/MFDeriv/Basic.lean:480]: Yes. I needed the `ContMDiff` version to add the deprecated alias.

**Comment by sgouezel** [comment]
>     * would you like to remove the now-used `abbrev`s `Smooth` etc?

Not sure if it's a question on the current version of the PR. In the current version of the PR, `Smooth` and friends are deprecated aliases of `ContMDiff` and friends, because they may be used by downstream projects so deprecations are in order.

>     * the deprecation dates in your lemmas are slightly inconsistent: not a big deal, in any case

Yes, it depends on the day I've made the change. Since it's just by one or two days, I wouldn't mind. Maybe one day CI will automagically change the deprecated dates to the date the PR is merged in master, but we're not there yet :-)

**Comment by winstonyin** [comment]
I looked through the changes. The deduplication of API is worth the occasional addition of `le_top`. Thanks for fixing the `Mdiff` casing!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**